Dockerfile

```diff
@@ -1,39 +1,78 @@
-FROM python:3.10.13-slim
+###################################################################################
+FROM  python:3.12-slim as base
 
-RUN adduser alkivi
+# Set apt variables to avoid interactive mode
+ENV  DEBIAN_FRONTEND=noninteractive
+# DL4006 info: Set the SHELL option -o pipefail before RUN with a pipe in it.
+SHELL  ["/bin/bash", "-o", "pipefail", "-c"]
 
-ENV PYTHONUNBUFFERED=1 \
-    PYTHONDONTWRITEBYTECODE=1 \
-    \
-    PIP_NO_CACHE_DIR=off \
-    PIP_DISABLE_PIP_VERSION_CHECK=on \
-    PIP_DEFAULT_TIMEOUT=100 \
-    \
-    POETRY_VERSION=1.4.2 \
-    POETRY_HOME="/opt/poetry" \
-    POETRY_VIRTUALENVS_IN_PROJECT=true \
-    POETRY_NO_INTERACTION=1 \
-    \
-    PYSETUP_PATH="/opt/pysetup" \
-    VENV_PATH="/opt/pysetup/.venv"
-
-RUN DEBIAN_FRONTEND=noninteractive apt-get -qq update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install -y \
+# Update the list of packages, install minimal packages
+RUN  apt-get update \
+    && apt-get install --no-install-recommends -y \
+    apt-utils \
     curl \
     libpcre3 \
     libpcre3-dev \
     libssl-dev \
-    build-essential
-RUN pip install "poetry==$POETRY_VERSION"
-
-COPY pyproject.toml poetry.lock ./
-RUN poetry export --dev --without-hashes --no-interaction --no-ansi -f requirements.txt -o requirements.txt
-RUN pip install -r requirements.txt
-COPY . /application
-WORKDIR /application
-RUN chmod a+x boot.sh
-
-RUN chown -R alkivi:alkivi ./
-USER alkivi
-EXPOSE 8000
-ENTRYPOINT ["./boot.sh"]
+    build-essential \
+    libjpeg62-turbo-dev \
+    zlib1g-dev
+# Clean apt to minimize size of image
+RUN  apt-get clean
+RUN  rm -rf /var/lib/apt/lists/*
+# Add default user for the docker container to be used as non-root
+RUN  adduser superquinquin
+
+###################################################################################
+FROM  base as pybase
+# Keeps Python from generating .pyc files in the container
+ENV  PYTHONDONTWRITEBYTECODE=1
+# Turns off buffering for easier container logging
+ENV  PYTHONUNBUFFERED=1
+# Configure PIP
+ENV  PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100
+# Configure Poetry
+ENV  POETRY_VERSION=1.7.1
+# DL4006 info: Set the SHELL option -o pipefail before RUN with a pipe in it.
+SHELL  ["/bin/bash", "-o", "pipefail", "-c"]
+USER superquinquin
+# Install poetry
+RUN  curl -sSL https://install.python-poetry.org | python3 - --version ${POETRY_VERSION}
+
+
+###################################################################################
+FROM pybase as final
+
+ARG  LABEL_NAME="printer"
+ARG  LABEL_VERSION="0.2.0"
+ARG  LABEL_URL="https://github.com/superquinquin/Brcd-Printer/"
+
+ENV  APP_DIR="/home/superquinquin/printer"
```

Comment:
I suppose volume path in the docker compose must change as well then.

Reply:
Check latest commit and I will merge if ok !
[fix(docker-compose.yml): change path for application volume](https://github.com/superquinquin/Brcd-Printer/pull/2/commits/300398fabcce8a815922ee9bb76371c65edefbf0)